tidy3d/web/api/autograd/autograd.py

```diff
@@ -54,37 +55,55 @@
 _INSPECT_ADJOINT_PLANE = td.Box(center=(0, 0, 0), size=(td.inf, td.inf, 0))
 
 
-def is_valid_for_autograd(simulation: td.Simulation) -> bool:
-    """Check whether a supplied simulation can use autograd run."""
+def is_valid_for_autograd(simulation: SimulationType) -> bool:
+    """Check whether a supplied object can use the autograd path.
 
-    # only support Simulations
-    if not isinstance(simulation, td.Simulation):
-        return False
-
-    # if no tracers just use regular web.run()
-    traced_fields = simulation._strip_traced_fields(
-        include_untraced_data_arrays=False, starting_path=("structures",)
-    )
-    if not traced_fields:
-        return False
+    Supports either a single td.Simulation or a ComponentModelerType. For a component
+    modeler, returns True if any of the underlying simulations are valid for autograd.
+    """
 
-    # if no frequency-domain data (e.g. only field time monitors), raise an error
-    if not simulation._freqs_adjoint:
-        raise AdjointError(
-            "No frequency-domain data found in simulation, but found traced structures. "
-            "For an autograd run, you must have at least one frequency-domain monitor."
+    def _is_valid_sim(sim: td.Simulation) -> bool:
```

Comment:
one potential thing we could add is a private property `_is_valid_for_autograd()` to a few classes, such as `Simulation` and `ComponentModeler` and then just call that here. That would separate out this logic specific to the component modeler